### PR TITLE
fix: avoid duplicate node builtin helper imports

### PR DESCRIPTION
**Pull request by lumirlumir**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/.github/blob/master/CONTRIBUTING.md).

#### AI acknowledgment

- [x] I did _not_ use AI to generate this PR.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

#### What is the purpose of this pull request?

This PR fixes an issue in `@eslint/v8-to-v9-config` where duplicate helper import detection was too narrow. The previous check only matched exact double-quoted specifiers like `from "url"` and `from "path"`, so configs that already used single quotes or `node:`-prefixed built-in modules could receive duplicate imports.

#### What changes did you make? (Give an overview)

This PR updates the duplicate import detection to recognize both quote styles and `node:` specifiers, while ensuring the opening and closing quotes match.

#### Related Issues

N/A

#### Is there anything you'd like reviewers to focus on?

Actually, the current fixture-based test approach can’t verify all edge cases in the tests. I’ll add a more robust one if the ongoing addition of the unit test runner is accepted.

### CODE REVIEW

File: .lintstagedrc.mjs
Comment (lumirlumir) [inline .lintstagedrc.mjs:1]: Updated the `lint-staged` config file to ensure `.md` files are also formatted and to avoid errors on unmatched patterns, as some files under the ignore patterns of `oxlint` or `oxfmt` cause errors here.

**Review by AugustinMauroy — APPROVED** [review]
LGMT maybe your utility can be refractored to use ast because your regex can be replaced by a query of specifier

**Comment by alexbit-codemod** [comment]
@lumirlumir Do u wanna push the fix as is and consider Augustin's feedback for future improvements? Or do u want to fix that as part of this PR?

File: .changeset/quiet-timers-sing.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@eslint/v8-to-v9-config': patch
+---
+
+Recognize single-quoted and `node:`-prefixed builtin module specifiers when adding `__dirname` helper imports, preventing duplicate `path` imports for configs that already require `node:path` while emitting `node:url` and `node:path` helpers.
```
[the remark this answers is not in the document]
Comment (lumirlumir) [reply .changeset/quiet-timers-sing.md]: Due to Copilot not having CLA permission, I force-pushed this change in https://github.com/eslint/codemods/pull/25/changes/f1019e54043767d8e5d44af884797bc6706716ba 😓

**Comment by lumirlumir** [comment]
> maybe your utility can be refractored to use ast because your regex can be replaced by a query of specifier

Thanks for the suggestion 🙂 I found that this kind of regex pattern approach is used elsewhere, so I hope it can be refactored together in a more robust way in a follow-up PR. I’ll open one soon.

https://github.com/eslint/codemods/blob/3cf58a5d79e3dff1e2cbf2649cbe3e86ae46bed6/codemods/v9/config-file/utils/make-new-config.ts#L466-L467

> Do u wanna push the fix as is and consider Augustin's feedback for future improvements? Or do u want to fix that as part of this PR?

Yep, I think it’s good to merge as is for now, and I’d like to find a more robust AST-based approach in a follow-up PR.

**Review by alexbit-codemod — APPROVED** [review]
